libkbfs/md_ops.go

```diff
@@ -20,6 +21,17 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+const (
+	// Our contract with the server states that it won't accept KBFS
+	// writes if more than 3 hours have passed since the last Merkle
+	// roots (both global and KBFS) were published.  Add some padding
+	// to that, and if we see any gaps larger than this, we will know
+	// we shouldn't be trusting the server.
+	maxAllowedMerkleGap = 3*time.Hour + 15*time.Minute
+
+	merkleGapEnforcementStartString = "2018-May-17"
```

Comment:
Perhaps use the RFC3339 format here so that it includes timezone?

Also, if we screw up mdmerkle at some point (especially while the write-ordering work is still on-going), is this what we dial forward to "unbreak" clients?

Reply:
> Perhaps use the RFC3339 format here so that it includes timezone?

Sure, will do.

> Also, if we screw up mdmerkle at some point (especially while the write-ordering work is still on-going), is this what we dial forward to "unbreak" clients?

Depends what you mean by "screw up".  I'll add a comment to clarify, but basically this should reflect when we deploy the mdserver version that rejects writes after it's been too long since merkle published.  It doesn't help any other kinds of screw-ups.

(I'll have to update this value to whatever the next mdserver prod deploy is.)

Reply:
Right; I meant deploy/ops screwup. Thanks for clearing up!